src/main/java/com/segment/analytics/android/integrations/optimizelyx/OptimizelyXIntegration.java

```diff
@@ -1,63 +1,102 @@
-import android.os.Bundle;
+import com.optimizely.ab.config.Experiment;
+import com.optimizely.ab.event.LogEvent;
 import com.segment.analytics.Analytics;
+import com.segment.analytics.Options;
+import com.segment.analytics.Traits;
 import com.segment.analytics.Properties;
 import com.segment.analytics.ValueMap;
-import com.segment.analytics.integrations.IdentifyPayload;
 import com.segment.analytics.integrations.Integration;
-import com.segment.analytics.integrations.Logger;
-import com.segment.analytics.integrations.TrackPayload;
-import java.util.List;
+
+import static com.segment.analytics.internal.Utils.isNullOrEmpty;
+
+import com.optimizely.ab.notification.NotificationListener;
+import com.optimizely.ab.config.Variation;
+
+import java.util.Map;
 
 /**
- * Kahuna helps mobile marketers send push notifications and in-app messages.
+ * Optimizely X helps marketers and growth hackers make smart decisions through A/B testing.
  *
- * @see <a href="https://www.optimizely.com/">Pptimizely</a>
- * @see <a href="https://segment.com/docs/integrations/optimizely/">Optimizely Integration</a>
- * @see <a href="http://developers.optimizely.com/android/">Optimizely Android SDK</a>
+ * @see <a href="https://www.optimizely.com/">Optimizely</a>
+ * @see <a href="https://segment.com/docs/destinations/optimizelyx/">Optimizely Integration</a>
+ * @see <a
+ *     href="https://developers.optimizely.com/x/solutions/sdks/introduction/index.html?language=android&platform=mobile">Optimizely
+ *     Android SDK</a>
  */
-public class OptimizelyXIntegration extends Integration<Void> {
-  private static final String OPTIMIZELY_KEY = "Optimizely";
-
-  private final Context context;
-  private final Logger logger;
-  private final Analytics analytics;
-
-  // Optimizely must be initialized immediately on launch. So we require the token when creating the
-  // factory, and initialize it in this method.
-//  public static Factory createFactory(final Application application, String token) {
-////    Optimizely.startOptimizelyWithAPIToken(token, application);
-////    return new Factory() {
-////      @Override public Integration<?> create(ValueMap settings, Analytics analytics) {
-////        boolean listen = settings.getBoolean("listen", false);
-////        Logger logger = analytics.logger(OPTIMIZELY_KEY);
-////
-////        OptimizelyIntegration integration =
-////            new OptimizelyIntegration(analytics, application, logger);
-////        if (listen) {
-////          Optimizely.addOptimizelyEventListener(integration);
-////        }
-////        return integration;
-////      }
-//
-////      @Override public String key() {
-////      return OPTIMIZELY_KEY;
-//////      }
-//    };
-//  }
-
-  public OptimizelyXIntegration(Analytics analytics, Context context, Logger logger) {
-    this.analytics = analytics;
-    this.context = context;
-    this.logger = logger;
-  }
+public class OptimizelyXIntegration extends Integration<NotificationListener> {
```

Comment:
Why should this return the listener? I think this should return `Void`.

Reply:
Should be something like `public class OptimizelyXIntegration extends Integration<Void> implements OptimizelyEventListener {` yeah?

Reply:
functionally yes. the exact code wouldn't work because `OptimizelyEventListener` is in the old version only; while the new version has `NotificationListener` it's an abstract class and not an interface, so it needs to be structured differently.